extension/js/common/platform/store/contact-store.ts

```diff
@@ -312,110 +387,108 @@ export class ContactStore extends AbstractStore {
     return str.normalize('NFKD').replace(/[\u0300-\u036F]/g, '').toLowerCase();
   }
 
-  private static dbIndex = (hasPgp: boolean, substring: string) => {
-    if (!substring) {
-      throw new Error('db_index has to include substring');
-    }
+  private static dbIndex = (hasPgp: boolean, substring: string): string => {
     return (hasPgp ? 't:' : 'f:') + substring;
   }
 
-  private static dbCreateSearchIndexList = (email: string, name: string | null, hasPgp: boolean) => {
-    email = email.toLowerCase();
-    name = name ? name.toLowerCase() : '';
-    const parts = [email, name];
-    const domain: string = email.split('@').pop() || '';
-    parts.push(domain, ...email.split(/[^a-z0-9]/));
-    parts.push(...name.split(/[^a-z0-9]/));
-    const index: string[] = [];
-    for (const part of parts.filter(p => !!p)) {
-      let substring = '';
-      for (const letter of part.split('')) {
-        substring += letter;
-        const normalized = ContactStore.normalizeString(substring);
-        if (!index.includes(normalized)) {
-          index.push(ContactStore.dbIndex(hasPgp, normalized));
-        }
-      }
+  private static dbIndexRange = (hasPgp: boolean, substring: string): { lowerBound: string, upperBound: string } => {
+    const lowerBound = ContactStore.dbIndex(hasPgp, substring);
+    let copyLength = lowerBound.length - 1;
+    let lastChar = lowerBound.charCodeAt(copyLength);
+    while (lastChar >= 65535) {
```

Comment:
Please add a code comment about what this line does and why

Reply:
Done